biochemist-python/chapters/Working_with_Pandas.ipynb

```diff
@@ -0,0 +1,1172 @@
+    "    \n",
+    "<p>Questions</p>\n",
+    "    <ul>\n",
+    "        <li>How can I use import data for analysis in my notebook?\n",
+    "    </ul>\n",
+    "<p>Objectives:</p>\n",
+    "    <ul>\n",
+    "        <li>Import the pandas library.\n",
+    "        <li>Use pandas library funtions to import data from a csv file.\n",
+    "        <li>Import data from a .csv formatted file.\n",
+    "    </ul>\n",
+    "<p>Keypoints:</p>\n",
+    "    <ul>\n",
+    "        <li>Use the pandas library to create dataframes from csv formatted data.</li>\n",
+    "    </ul>\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## What is pandas and why do we use it?\n",
+    "\n",
+    "Pandas is a python library that is designed to work with two dimensional data arrays. It is built on numpy, another python library that specializes in numerical analysis. Numpy also has the ability to create and analyze data arrays. If you are not familiar with arrays, here are some simple examples.\n",
+    "\n",
+    "### 1D arrays\n",
+    "A one dimensional array is simply a list of items, for example, a list of the elements: H, He, Li, Be, B, etc.\n",
+    "\n",
+    "### 2D arrays\n",
+    "A two dimensional array contains matched pairs of information. A list of elements with their atomic numbers would be a 2D array:  (H, 1), (He, 2), (Li, 3), (Be, 4), (B, 5), etc. It could also be a list of x, y data points that you want to plot: (1,1), (2,3), (3,5), (4,7). \n",
+    "\n",
+    "### 3D arrays\n",
+    "A three dimensional array contains matched triads of information. One common 3D array in biochemistry and structural biology would be a collection of x, y, and z coordinates for the atoms in a protein, as are found in the [Protein Data Bank](http://www.rcsb.org/pdb). \n",
+    "\n",
+    "The numpy library has functions that will manage n-dimensional arrays, while pandas only works on 2D arrays. You may need numpy for nD arrays at some point, but for this workshop, we will learn to use pandas as we build toward the ability to perform linear and nonlinear regression of laboratory data based on data in 2D arrays.\n",
+    "\n",
+    "The pandas library contains powerful tools for working with 2D data arrays, including the ability to identify the rows and columns of data by unique identifiers: things like \"mass protein\" and \"initial velocity\" rather than \"row 1\" or \"column C\"). Pandas also has many more functions that we will not explore in this workshop, but here are two excellent free online resources for learning more about pandas:\n",
+    "\n",
+    "+ Charlie Weiss's excellent online textbook, *Scientific Computing for Chemists*, which you can find on his GitHub site, [SciCompforChemists](https://github.com/weisscharlesj/SciCompforChemists)\n",
+    "+ Corey Schafer's [Pandas Tutorials on YouTube](https://www.youtube.com/watch?v=ZyhVh-qRZPA&list=PL-osiE80TeTsWmV9i9c58mdDCSskIFdDS)"
```

Comment:
Add the MolSSI lesson on pandas here - https://education.molssi.org/python-data-analysis/02-pandas/index.html

I've mentioned this lesson before but I've noticed you never reference it. Why not? What could be done to make this lesson more useful?

Reply:
I'm not sure that I have ever looked at it before. I should have.